### PR TITLE
Rush Tank

### PR DESCRIPTION
**Pull request by zhqwqwq**
增加了一种新的坦克【Rush Tank】。其skill为增加移动速度，持续时间三秒。

### CODE REVIEW

**Review by qinchuanhui — CHANGES_REQUESTED** [review]
The same problem as #352 , plz re-organize ur code first

File: src/battle_game/core/units/shadow_tank_zhq.h [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+#include "battle_game/core/units/tiny_tank.h"
+
+namespace battle_game::unit {
+class ShadowTank : public Tank {
+ public:
+  ShadowTank(GameCore *game_core, uint32_t id, uint32_t player_id);
+  void Update() override;
+
+ protected:
+  ;
```
Comment (qinchuanhui) [inline src/battle_game/core/units/shadow_tank_zhq.h:12]: what‘s this

File: src/battle_game/core/units/rush_tank_zhq.cpp [hunk 2]
```diff
@@ -0,0 +1,95 @@
+#include <ctime>
+
+#include "battle_game/core/game_core.h"
+
+namespace battle_game::unit {
+
+namespace {
+uint32_t tank_body_model_index = 0xffffffffu;
+uint32_t tank_turret_model_index = 0xffffffffu;
+}  // namespace
```
Comment (qinchuanhui) [inline src/battle_game/core/units/rush_tank_zhq.cpp:10]: seems u don't use these? maybe just delete them

File: src/battle_game/core/units/rush_tank_zhq.h [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  RushTank(GameCore *game_core, uint32_t id, uint32_t player_id);
+  void Update() override;
+
+ protected:
+  void TankMove(float move_speed, float rotate_angular_speed) override;
+  [[nodiscard]] const char *UnitName() const override;
+  [[nodiscard]] const char *Author() const override;
+  void Rush();
+  void RushClick();
+  int time_stamp;
```
Comment (qinchuanhui) [inline src/battle_game/core/units/rush_tank_zhq.h:17]: `time_stamp_`

**Review by qinchuanhui — CHANGES_REQUESTED** [review]

**Review by qinchuanhui — APPROVED** [review]
LGTM, you can get the corresponding score

**Comment by zhqwqwq** [comment]
This PR is for issue #334

**Pull request closed without merging**